deploy/admission_webhook.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: gateway-api
+  labels:
+    name: gateway-api
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: gateway-api-admission
+webhooks:
+  - name: validate.networking.x-k8s.io
+    matchPolicy: Equivalent
+    rules:
+      - operations: [ "CREATE" , "UPDATE" ]
+        apiGroups: [ "networking.x-k8s.io" ]
+        apiVersions: [ "v1alpha1" ]
+        resources: [ "httproutes" ]
+    failurePolicy: Fail
+    sideEffects: None
+    admissionReviewVersions:
+      - v1
+    clientConfig:
+      service:
+        name: gateway-api-admission-server
+        namespace: gateway-api
+        path: "/validate"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: gateway-api-webhook-server
+    version: 0.0.1
+  name: gateway-api-admission-server
+  namespace: gateway-api
+spec:
+  type: ClusterIP
+  ports:
+    - name: https-webhook
+      port: 443
+      targetPort: 8443
+  selector:
+    name: gateway-api-admission-server
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gateway-api-admission-server
+  namespace: gateway-api
+  labels:
+    name: gateway-api-admission-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: gateway-api-admission-server
+  template:
+    metadata:
+      name: gateway-api-admission-server
+      labels:
+        name: gateway-api-admission-server
+    spec:
+      containers:
+        - name: webhook
+          # TODO(hbagdi): Swap image name to the k8s official image
+          image: hbagdi/gateway-api-admission-webhook:latest
```

Comment:
I'm working on getting us a new docker repo to host this.

Reply:
Sorry, missed that this was before the Gateway API rename, thanks for following up with https://github.com/kubernetes/k8s.io/pull/1954!

Reply:
Is it cool to merge this in before we get the image repo set up?
It feels like this is going to be a chicken and egg problem which will require some iterations.

Reply:
@robscott What if I update the `image` to `TODO` and then work on another PR to get [cloudbuild](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md) setup for admission webhook?

Reply:
I think that's reasonable, adding to agenda for community meeting